crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -38,17 +37,17 @@ impl ProgrammableTransactionBuilder {
 
     pub fn finish(self) -> ProgrammableTransaction {
         let Self { inputs, commands } = self;
-        let inputs = inputs.into_values().collect();
+        let inputs = inputs.into_iter().collect();
         ProgrammableTransaction { inputs, commands }
     }
 
-    pub fn pure_bytes(&mut self, bytes: Vec<u8>, force_separate: bool) -> Argument {
-        let arg = if force_separate {
-            BuilderArg::ForcedNonUniquePure(self.inputs.len())
-        } else {
-            BuilderArg::Pure(bytes.clone())
-        };
-        let (i, _) = self.inputs.insert_full(arg, CallArg::Pure(bytes));
+    pub fn pure_bytes(&mut self, bytes: Vec<u8>, _force_separate: bool) -> Argument {
+        // let arg = if force_separate {
+        //     BuilderArg::ForcedNonUniquePure(self.inputs.len())
+        // } else {
+        //     BuilderArg::Pure(bytes.clone())
+        // };
+        let (i, _) = self.inputs.insert_full(CallArg::Pure(bytes));
```

Comment:
It looks like `IndexMap` is necessary precisely to support this feature -- preventing the merging of pure inputs that have the same contents.